pkg/distro/fedora/imagetype.go

```diff
@@ -151,6 +151,24 @@ func (t *imageType) getPartitionTable(
 	}
 
 	imageSize := t.Size(options.Size)
+	partitioning := customizations.GetPartitioning()
+	if partitioning != nil {
+		// Use the new custom partition table to create a PT fully based on the user's customizations.
+		// This overrides FilesystemCustomizations, but we should never have both defined.
+		if options.Size > 0 {
+			// user specified a size on the command line, so let's override the
+			// customization with the calculated/rounded imageSize
+			partitioning.MinSize = imageSize
+		}
+
+		partOptions := &disk.CustomPartitionTableOptions{
+			PartitionTableType: basePartitionTable.Type, // copy from image type base
+			BootMode:           t.BootMode(),
+			DefaultFSType:      disk.FS_EXT4, // default fs type for Fedora (when not btrfs)
```

Comment:
> Maybe also worth a small comment that the default fs is not customizable currently (which is curious now that I think about it because the user can de-facto have a default FS by just specifiying all mount points manually and adding their preferred FStype - I forgot if we ever talked about this).

Yes, the user can specify all partitions and their fs type.  There's no way for them to specify what the root filesystem should be if it's automatically added however.  I think as a customization it's kinda pointless: why have a customization for changing the fs type of the automatically created `/` when it's the same as simply specifying `{"mountpoint": "/", "type": "whatever"}`.  So this is a distro-wide policy of "default fs type", which is usually ext4 for Fedora and xfs for RHEL/CentOS.

> Also maybe worth clarifying a bit more what "(when not btrfs)" means, maybe something like `// ext4 is the default on fedora for all non-btrfs partitions, btrfs is modeled as a partition type not a filesystem type here` (or something, I'm not happy with my comment, it feels too long).

eh, I'd rather just remove the comment or keep "ext4 is the default for Fedora".  The "when not btrfs" was more of a clarification for the mechanics of automatic subvolumes being created on btrfs, but I think that doesn't need clarification tbh.